core/arch/arm/kernel/thread_a64.S

```diff
@@ -507,6 +529,117 @@ el0_serror_a32:
 	b	el0_serror_a32
 	check_vector_size el0_serror_a32
 
+#if defined(CFG_CORE_WORKAROUND_SPECTRE_BP_SEC)
+	.macro apply_workaround
```

Comment:
`apply_workaround` is a bit too vague IMO. `s/apply_workaround/ic_iallu_isb/`? Or if you prefer `s/apply_workaround/icache_invalidate/`?

Reply:
ARM-TF is mentioning a low overhead SMC interface for to invalidate the branch predictor, maybe that could be used later.
I guess `s/apply_workaround/invalidate_branch_predictor/` would be good enough.

Reply:
Sounds good